session.go

```diff
@@ -298,6 +298,32 @@ func newSenderForSession(ctx context.Context, s *Session, target string, opts *S
 	return l, nil
 }
 
+// NewTransactionController creates a transaction controller, which can declare
+// and dischage transactions for this connection.
```

Comment:
`for this Session`?

Reply:
It's not just for this session, it can be used connection-wide.

Reply:
If it's not scoped to a session then shouldn't `NewTransactionCoordinator` be moved to `Conn`? Where it's at now implies it's unique per `Session`.

Reply:
I considered this but it's more of a feature of this that you can just create a transaction controller if you have a session (ie, it's just a link). 

I ended up just thinking it's simpler, design-wise, to keep it as a sibling of the other two types and just treat it like a simpler Sender.

Reply:
I get that it's just a link (which I would contend is an implementation detail) but if it's connection-wide then conceptually, it makes sense to have it at the connection level. In addition, this means that the transaction coordinator would have its own session, so it doesn't have to share the incoming/outgoing window with other links in the same session or be torn down if we recycle the session for a sender/receiver.

Reply:
Looking at this some more I think the connection affinity that I've mentioned is a broker detail (ie, probably SB specific). So I really think what we've got is best since it just maps to the basic AMQP details.

http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#type-txn-capability

Reply:
(and to add another thing I just found)

https://github.com/Azure/azure-sdk-for-java/pull/19863

Cross-entity transactions, I believe, are an SB specific feature, but it does require more control over what is created with what AMQP session. 

So I think we're on the right track just treating this as another entity, and letting the user fully control the session.